gate-integrations-gremlin/src/main/groovy/com/netflix/spinnaker/gate/config/gremlin/GremlinGateConfig.groovy

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2014 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.gate.config.gremlin
+
+import com.fasterxml.jackson.databind.DeserializationFeature
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.netflix.spectator.api.Registry
+import com.netflix.spinnaker.gate.config.Service
+import com.netflix.spinnaker.gate.config.ServiceConfiguration
+import com.netflix.spinnaker.gate.retrofit.EurekaOkClient
+import com.netflix.spinnaker.gate.retrofit.Slf4jRetrofitLogger
+import com.netflix.spinnaker.gate.services.EurekaLookupService
+import com.netflix.spinnaker.gate.services.gremlin.GremlinService
+import com.squareup.okhttp.OkHttpClient
+import groovy.transform.CompileStatic
+import groovy.util.logging.Slf4j
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import retrofit.Endpoint
+import retrofit.RequestInterceptor
+import retrofit.RestAdapter
+import retrofit.converter.JacksonConverter
+
+import static retrofit.Endpoints.newFixedEndpoint
+
+@CompileStatic
+@Configuration
+@Slf4j
+class GremlinGateConfig {
+
+  @Value('${retrofit.logLevel:BASIC}')
+  String retrofitLogLevel
+
+  @Autowired
+  ServiceConfiguration serviceConfiguration
+
+  @Autowired
+  Registry registry
+
+  @Autowired
+  EurekaLookupService eurekaLookupService
+
+  @Autowired
+  RequestInterceptor spinnakerRequestInterceptor
+
+  @Bean
+  @ConditionalOnProperty('integrations.gremlin.enabled')
+  GremlinService gremlinService(OkHttpClient okHttpClient) {
+    createClient('gremlin', GremlinService, okHttpClient)
+  }
+
+  private <T> T createClient(String serviceName,
```

Comment:
These helper methods are copied from `GateConfig`.  A more DRY approach would be to have some abstracted-out code that both `GateConfig` and `GremlinGateConfig` depend on, but I wanted to get your input before I attempt to refactor one of the core classes